// transbank-sdk/dist/es5/transbank/common/api_constants.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
/**
 * Contains the Webpay and Oneclick URLs base.
 */
var ApiConstants;
(function (ApiConstants) {
    ApiConstants["WEBPAY_ENDPOINT"] = "/rswebpaytransaction/api/webpay/v1.2";
    ApiConstants["ONECLICK_ENDPOINT"] = "/rswebpaytransaction/api/oneclick/v1.2";
    ApiConstants["PATPASS_COMERCIO_ENDPOINT"] = "/restpatpass/v1/services";
    ApiConstants[ApiConstants["BUY_ORDER_LENGTH"] = 26] = "BUY_ORDER_LENGTH";
    ApiConstants[ApiConstants["SESSION_ID_LENGTH"] = 61] = "SESSION_ID_LENGTH";
    ApiConstants[ApiConstants["RETURN_URL_LENGTH"] = 255] = "RETURN_URL_LENGTH";
    ApiConstants[ApiConstants["AUTHORIZATION_CODE_LENGTH"] = 6] = "AUTHORIZATION_CODE_LENGTH";
    ApiConstants[ApiConstants["CARD_EXPIRATION_DATE_LENGTH"] = 5] = "CARD_EXPIRATION_DATE_LENGTH";
    ApiConstants[ApiConstants["CARD_NUMBER_LENGTH"] = 19] = "CARD_NUMBER_LENGTH";
    ApiConstants[ApiConstants["TBK_USER_LENGTH"] = 40] = "TBK_USER_LENGTH";
    ApiConstants[ApiConstants["USER_NAME_LENGTH"] = 40] = "USER_NAME_LENGTH";
    ApiConstants[ApiConstants["COMMERCE_CODE_LENGTH"] = 12] = "COMMERCE_CODE_LENGTH";
    ApiConstants[ApiConstants["TOKEN_LENGTH"] = 64] = "TOKEN_LENGTH";
    ApiConstants[ApiConstants["EMAIL_LENGTH"] = 100] = "EMAIL_LENGTH";
})(ApiConstants || (ApiConstants = {}));
exports.default = ApiConstants;
