// transbank-sdk/dist/es5/transbank/common/integration_commerce_codes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
/**
 * Contains the Webpay, Oneclick and Patpass Comercio constants for testing.
 */
var IntegrationCommerceCodes;
(function (IntegrationCommerceCodes) {
    IntegrationCommerceCodes["WEBPAY_PLUS"] = "597055555532";
    IntegrationCommerceCodes["WEBPAY_PLUS_MODAL"] = "597055555584";
    IntegrationCommerceCodes["WEBPAY_PLUS_DEFERRED"] = "597055555540";
    IntegrationCommerceCodes["WEBPAY_PLUS_MALL"] = "597055555535";
    IntegrationCommerceCodes["WEBPAY_PLUS_MALL_CHILD1"] = "597055555536";
    IntegrationCommerceCodes["WEBPAY_PLUS_MALL_CHILD2"] = "597055555537";
    IntegrationCommerceCodes["WEBPAY_PLUS_MALL_DEFERRED"] = "597055555581";
    IntegrationCommerceCodes["WEBPAY_PLUS_MALL_DEFERRED_CHILD1"] = "597055555582";
    IntegrationCommerceCodes["WEBPAY_PLUS_MALL_DEFERRED_CHILD2"] = "597055555583";
    IntegrationCommerceCodes["ONECLICK_MALL"] = "597055555541";
    IntegrationCommerceCodes["ONECLICK_MALL_CHILD1"] = "597055555542";
    IntegrationCommerceCodes["ONECLICK_MALL_CHILD2"] = "597055555543";
    IntegrationCommerceCodes["ONECLICK_MALL_DEFERRED"] = "597055555547";
    IntegrationCommerceCodes["ONECLICK_MALL_CHILD1_DEFERRED"] = "597055555548";
    IntegrationCommerceCodes["ONECLICK_MALL_CHILD2_DEFERRED"] = "597055555549";
    IntegrationCommerceCodes["TRANSACCION_COMPLETA"] = "597055555530";
    IntegrationCommerceCodes["TRANSACCION_COMPLETA_SIN_CVV"] = "597055555557";
    IntegrationCommerceCodes["TRANSACCION_COMPLETA_DEFERRED"] = "597055555531";
    IntegrationCommerceCodes["TRANSACCION_COMPLETA_DEFERRED_SIN_CVV"] = "597055555556";
    IntegrationCommerceCodes["TRANSACCION_COMPLETA_MALL"] = "597055555573";
    IntegrationCommerceCodes["TRANSACCION_COMPLETA_MALL_CHILD1"] = "597055555574";
    IntegrationCommerceCodes["TRANSACCION_COMPLETA_MALL_CHILD2"] = "597055555575";
    IntegrationCommerceCodes["TRANSACCION_COMPLETA_MALL_SIN_CVV"] = "597055555551";
    IntegrationCommerceCodes["TRANSACCION_COMPLETA_MALL_SIN_CVV_CHILD1"] = "597055555552";
    IntegrationCommerceCodes["TRANSACCION_COMPLETA_MALL_SIN_CVV_CHILD2"] = "597055555553";
    IntegrationCommerceCodes["TRANSACCION_COMPLETA_MALL_DEFERRED"] = "597055555576";
    IntegrationCommerceCodes["TRANSACCION_COMPLETA_MALL_DEFERRED_CHILD1"] = "597055555577";
    IntegrationCommerceCodes["TRANSACCION_COMPLETA_MALL_DEFERRED_CHILD2"] = "597055555578";
    IntegrationCommerceCodes["TRANSACCION_COMPLETA_MALL_DEFERRED_SIN_CVV"] = "597055555561";
    IntegrationCommerceCodes["TRANSACCION_COMPLETA_MALL_DEFERRED_SIN_CVV_CHILD1"] = "597055555562";
    IntegrationCommerceCodes["TRANSACCION_COMPLETA_MALL_DEFERRED_SIN_CVV_CHILD2"] = "597055555563";
    IntegrationCommerceCodes["PATPASS_COMERCIO"] = "28299257";
    IntegrationCommerceCodes["PATPASS_BY_WEBPAY"] = "597055555550";
})(IntegrationCommerceCodes || (IntegrationCommerceCodes = {}));
exports.default = IntegrationCommerceCodes;
