// transbank-sdk/dist/es5/transbank/webpay/webpay_plus/requests/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MallCaptureRequest = exports.MallRefundRequest = exports.MallCreateRequest = exports.CaptureRequest = exports.RefundRequest = exports.StatusRequest = exports.CommitRequest = exports.CreateRequest = void 0;
var create_request_1 = require("./create_request");
Object.defineProperty(exports, "CreateRequest", { enumerable: true, get: function () { return create_request_1.CreateRequest; } });
var commit_request_1 = require("./commit_request");
Object.defineProperty(exports, "CommitRequest", { enumerable: true, get: function () { return commit_request_1.CommitRequest; } });
var status_request_1 = require("./status_request");
Object.defineProperty(exports, "StatusRequest", { enumerable: true, get: function () { return status_request_1.StatusRequest; } });
var refund_request_1 = require("./refund_request");
Object.defineProperty(exports, "RefundRequest", { enumerable: true, get: function () { return refund_request_1.RefundRequest; } });
var capture_request_1 = require("./capture_request");
Object.defineProperty(exports, "CaptureRequest", { enumerable: true, get: function () { return capture_request_1.CaptureRequest; } });
var mall_create_request_1 = require("./mall_create_request");
Object.defineProperty(exports, "MallCreateRequest", { enumerable: true, get: function () { return mall_create_request_1.MallCreateRequest; } });
var mall_refund_request_1 = require("./mall_refund_request");
Object.defineProperty(exports, "MallRefundRequest", { enumerable: true, get: function () { return mall_refund_request_1.MallRefundRequest; } });
var mall_capture_request_1 = require("./mall_capture_request");
Object.defineProperty(exports, "MallCaptureRequest", { enumerable: true, get: function () { return mall_capture_request_1.MallCaptureRequest; } });
