// transbank-sdk/dist/es5/transbank/webpay/oneclick/requests/authorize_request.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.AuthorizeRequest = void 0;
var api_constants_1 = __importDefault(require("../../../common/api_constants"));
var request_base_1 = __importDefault(require("../../../common/request_base"));
var AuthorizeRequest = /** @class */ (function (_super) {
    __extends(AuthorizeRequest, _super);
    function AuthorizeRequest(userName, tbkUser, buyOrder, details) {
        var _this = _super.call(this, "".concat(api_constants_1.default.ONECLICK_ENDPOINT, "/transactions"), 'POST') || this;
        _this.userName = userName;
        _this.tbkUser = tbkUser;
        _this.buyOrder = buyOrder;
        _this.details = details;
        return _this;
    }
    AuthorizeRequest.prototype.toJson = function () {
        return JSON.stringify({
            username: this.userName,
            tbk_user: this.tbkUser,
            buy_order: this.buyOrder,
            details: this.details.map(function (detail) { return detail.toPlainObject(); }),
        });
    };
    return AuthorizeRequest;
}(request_base_1.default));
exports.AuthorizeRequest = AuthorizeRequest;
