// transbank-sdk/dist/es5/transbank/webpay/oneclick/requests/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CaptureRequest = exports.StatusRequest = exports.StartRequest = exports.RefundRequest = exports.FinishRequest = exports.DeleteRequest = exports.AuthorizeRequest = void 0;
var authorize_request_1 = require("./authorize_request");
Object.defineProperty(exports, "AuthorizeRequest", { enumerable: true, get: function () { return authorize_request_1.AuthorizeRequest; } });
var delete_request_1 = require("./delete_request");
Object.defineProperty(exports, "DeleteRequest", { enumerable: true, get: function () { return delete_request_1.DeleteRequest; } });
var finish_request_1 = require("./finish_request");
Object.defineProperty(exports, "FinishRequest", { enumerable: true, get: function () { return finish_request_1.FinishRequest; } });
var refund_request_1 = require("./refund_request");
Object.defineProperty(exports, "RefundRequest", { enumerable: true, get: function () { return refund_request_1.RefundRequest; } });
var start_request_1 = require("./start_request");
Object.defineProperty(exports, "StartRequest", { enumerable: true, get: function () { return start_request_1.StartRequest; } });
var status_request_1 = require("./status_request");
Object.defineProperty(exports, "StatusRequest", { enumerable: true, get: function () { return status_request_1.StatusRequest; } });
var capture_request_1 = require("./capture_request");
Object.defineProperty(exports, "CaptureRequest", { enumerable: true, get: function () { return capture_request_1.CaptureRequest; } });
